cpp/src/arrow/compute/exec/options.cc

```diff
@@ -61,5 +61,16 @@ Result<std::shared_ptr<SourceNodeOptions>> SourceNodeOptions::FromTable(
       new SourceNodeOptions(table.schema(), batch_gen));
 }
 
+Result<std::shared_ptr<SourceNodeOptions>> SourceNodeOptions::FromRecordBatchReader(
+    std::shared_ptr<RecordBatchReader> reader, std::shared_ptr<Schema> schema,
+    arrow::internal::Executor* exc) {
```

Comment:
Just a nit, but `executor` is used elsewhere in the codebase.
```suggestion
    arrow::internal::Executor* executor) {
```

Reply:
I copied the original signature from the previous `SourceNodeOptions::FromTable` which had `exec`. I renamed the parameter in both places.